pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go

```diff
@@ -191,16 +257,30 @@ func E2ETKRResolverValidationForClusterCRUDSpec(context context.Context, inputGe
 			Namespace:   namespace,
 			TkrVersion:  input.E2EConfig.TkrVersion,
 			SkipPrompt:  true,
-			Timeout:     timeout,
 		})
 		Expect(err).To(BeNil())
 
 		By(fmt.Sprintf("Validating the kubernetes version after cluster %q is upgraded", clusterName))
-		validateKubernetesVersion(clusterName, defaultTKR.Spec.Kubernetes.Version)
+		validateKubernetesVersion(clusterName, defaultTKR.Spec.Kubernetes.Version, tempKubeConfigFilePath)
 
 		By(fmt.Sprintf("Validating the TKR data after cluster %q is upgraded", clusterName))
 		verifyTKRData(context, mcProxy, options.ClusterName, options.Namespace)
 
+		// verify addons are deployed successfully in clusterclass mode
+		if isClusterClass, ok := input.OtherConfigs["clusterclass"]; ok && isClusterClass == "true" {
+			By(fmt.Sprintf("Verify addon packages on management cluster %q matches clusterBootstrap info on management cluster %q", input.E2EConfig.ManagementClusterName, input.E2EConfig.ManagementClusterName))
```

Comment:
nit: maybe considering add word like "after cluster upgrade" to differentiate with the check before the upgrade.

Reply:
Could I record this and make this change in a subsequent PR? the pipeline takes a long time to run.